src/Compilers/CSharp/Test/Semantic/Semantics/BindingAsyncTests.cs

```diff
@@ -3261,14 +3258,21 @@ async Task M1(bool truth)
         var tr = new TypedReference();
         await Task.Factory.StartNew(() => { });
     }
+
+    async Task M2(bool truth)
+    {
+        var tr = new TypedReference();
+        await Task.Factory.StartNew(() => { });
+        var tr2 = tr;
+    }
 }";
```

Comment:
>}" [](http://example.com/codeflow?start=0&length=2)

According to the speclet:
> the error is driven by the following mechanism: after an await expression (§12.9.8) or a yield return statement (§13.15), all ref local variables and variables of a ref struct type in scope are considered definitely unassigned (§9.4).

Is that already implemented? Do we allow the case where the local scope spans an `await` but the value is never used? For instance:
```csharp
async Task M3()
{
    var tr = new TypedReference();
    await Task.Factory.StartNew(() => { });
    tr = default;
    var tr2 = tr;
}

Reply:
> Is that already implemented?

Yes, that should be what IteratorAndAsyncCaptureWalker does.

> Do we allow the case where the local scope spans an await but the value is never used?

Yes, I will add your example to the test, thanks.